packages/core/src/external/surescripts/fhir/coverage.ts

```diff
@@ -1,47 +1,100 @@
 import { uuidv7 } from "@metriport/shared/util/uuid-v7";
-import { Coverage, Identifier } from "@medplum/fhirtypes";
+import { Coverage, Organization, Identifier, Reference } from "@medplum/fhirtypes";
 import { ResponseDetail } from "../schema/response";
-import { getPlanCodeName } from "@metriport/shared/interface/external/surescripts/plan-code";
+import {
+  getSourceOfPaymentCode,
+  getSourceOfPaymentName,
+} from "@metriport/shared/interface/external/surescripts/payment-code";
 import { getSurescriptsDataSourceExtension } from "./shared";
 import { SurescriptsContext } from "./types";
 import { getPatientReference } from "./patient";
 import {
   NCPDP_PROVIDER_ID_SYSTEM,
   PLAN_NETWORK_BIN_SYSTEM,
   PLAN_NETWORK_PCN_SYSTEM,
-  COVERAGE_TYPE_SYSTEM,
+  SOURCE_OF_PAYMENT_TYPOLOGY_SYSTEM,
 } from "./constants";
 
+export function getInsuranceOrganization(detail: ResponseDetail): Organization | undefined {
+  const paymentCode = detail.paymentCode ?? detail.planCode;
+  if (!paymentCode) return undefined;
+
+  const sourceOfPaymentCode = getSourceOfPaymentCode(paymentCode);
```

Comment:
This is the standard code according to the Source Of Payment Type concept mapping: `urn:oid:2.16.840.1.113883.3.221.5` - allowing for an organization <-> coverage link which mimics what is currently in the bundle as produced by the FHIR converter.